presto-orc/src/main/java/com/facebook/presto/orc/reader/ByteSelectiveStreamReader.java

```diff
@@ -312,10 +335,10 @@ else if (presentStream != null) {
 
     private void ensureValuesCapacity(int capacity, boolean recordNulls)
     {
-        values = ensureCapacity(values, capacity);
+        values = ensureCapacity(values, (int) ceil(capacity / (1 - BATCH_EXECUTION_FILTERRATE_THRESHOLD)));
```

Comment:
What's the rational for this logic?

Reply:
@mbasmanova thanks for reviewing the code! This change expands the values buffer based on the BATCH_EXECUTION_FILTERRATE_THRESHOLD. For example, if BATCH_EXECUTION_FILTERRATE_THRESHOLD = 0.3f, it means batch read for no filter case will be enabled when the filter rate for previous columns is below 30%, ie. if more than 30% rows were filtered out from previous columns, we will not do batch read. If there were less than 30% rows being filtered out from previous columns, we will do batch read. For batch read, we need to read more data (maxPositionCount) into the values buffer, so we will ensure its capacity is greater than 1/(1-0.3) * positionCount, which is 1.42x positionCount. Since these buffers are quite small, this extra space requirement is worthwhile as I showed that the gain could be up to 4x.

Reply:
@yingsu00 I suggest to keep this method as is and change the caller instead. This logic makes sense only for one branch of the code, but it applied to all branches. That's confusing.